Adjust and expand `safeinput` unit tests to follow issue #35 example more closely

jonasbardino opened the pull request:
Restructure `safeinput` unit tests to be more like recent `base` module unit tests and expand tests to explicitly cover the code path underpinning the issue #35 example.
Tighten asserts in existing tests. Fix some copy/paste comments and add some missing docstrings. Clean up unused imports.

jonasbardino committed:
Restructure safeinput unit tests to be more like recent base module unit tests
and expand tests to explicitly cover the code path underpinning the issue 35
example.
Tighten asserts in existing tests. Fix some copy/paste comments and add some
missing docstrings. Clean up unused imports.

jonasbardino committed:
Fix copy/paste error.

jonasbardino committed:
Fix some more copy/paste issues and add a few basic tests of the main helpers.

jonasbardino committed:
Added a few more simple tests of core functions.

jonasbardino committed:
Don't assign implicit None return value to local variable as lint will flag it.

albu-diku commented on tests/test_mig_shared_safeinput.py:
```diff
@@ -111,6 +125,102 @@ def test_commonname_filter(self):
+    def test_valid_commonname_accent_handling(self):
+        """Test accented character handling validation"""
+        # Common accents should pass with COMMON_ACCENTED
+        valid_commonname(self.ACCENTED_VALID)
```
I ound myself momentatily quit confused by this, until I realised these function are all throwing and thus what this is realy checking is that we do _not_ throw.

How can we communicate this to the reader? One thing might be defnining something that allows `with self.assertDoesNotRaise(): ...`, likely my preference, but if you don't wish to do that perhaps noting this in the test block docstring somewhere.

jonasbardino replied:
Well, yes the logical equivalent of `self.assertRaises` would be something like that.
I don't mind going that way if you prefer, but to clarify I've just wrapped them in `try/except` with an assert checking that we did _not_ end up in the except handler - like we already did in `test_commonname_valid`.

albu-diku reviewed (commented):
Generally fine - one comment to be considered and then happy to approve.

jonasbardino committed:
Address PR review feedback (thanks @albu-diku):
 - remove no-op `before_each` helper to limit confusion and noise
 - wrap the `valid_X` function calls we expect NOT to throw exception in
   try/except with assert to clarify check that they didn't throw exception.

jonasbardino commented on tests/test_mig_shared_safeinput.py:
```diff
@@ -84,7 +88,16 @@ def record_last_print(value):
+        """Provide test configuration"""
+        return 'testconfig'
+
+    def before_each(self):
```
Yes, removed now.

jonasbardino commented:
> Generally fine - one comment to be considered and then happy to approve.

Thanks for reviewing. I've addressed your comments.
